pandas/tests/groupby/test_raises.py

```diff
@@ -228,6 +271,17 @@ def test_groupby_raises_string_np(
             "Could not convert string .* to numeric",
         ),
     }[groupby_func_np]
+
+    if using_infer_string:
+        klass = TypeError
+        if df["d"].dtype.storage == "python":
+            msg = "Cannot perform reduction 'mean' with string dtype"
```

Comment:
The `'mean'` needs to be filled in based on `how` or `groupby_func_np.__name__`?